From df780e22a5898df4a0f52ceb0ae3579ea7c78646 Mon Sep 17 00:00:00 2001
From: Gavin Smith <gavinsmith0123@gmail.com>
Date: Fri, 8 Apr 2022 17:27:00 +0100
Subject: [PATCH] Msys2 OS detection in texi2dvi

* util/texi2dvi: Do not set MSYSTEM via OSTYPE environment
variable before calling uname, as this is not working on
Msys2.  Reverses change on 2015-09-23.  Instead, check the value of
OSTYPE directly as well as calling uname.

The change was stripped down removing the Changelog update and update
of texi2dvi version.

diff --git a/util/texi2dvi b/util/texi2dvi
index 1f42b41907..84ddde517c 100755
--- a/util/texi2dvi
+++ b/util/texi2dvi
@@ -80,12 +80,9 @@ IFS="$space$tab$newline"
 # directories in TEXINPUTS -- except for Cygwin and Msys, where COMSPEC
 # might be inherited, but : is used.
 
-# In the case of Msys, uname returns a value derived from MSYSTEM, as
-# MSYSTEM is user configurable, it is not so safe to use it to detect
-# Msys. It is safer to use OSTYPE, this is why we set MSYSTEM to
-# $OSTYPE before calling uname
 if test -n "$COMSPEC$ComSpec" \
-   && MSYSTEM=$OSTYPE uname | $EGREP -iv 'cygwin|msys' >/dev/null; then
+   && uname | $EGREP -iv 'cygwin|msys' >/dev/null \
+   && test "$OSTYPE" != msys ; then
   path_sep=";"
 else
   path_sep=":"

